episodes/08-crop-raster-data.md

```diff
@@ -36,259 +36,189 @@ We also use the cropped fields polygons `fields_cropped.shp`, which was generate
 ```
 
+![](fig/E08/visual_large.png){alt="Large visual raster"}
 
-## Crop raster data with a bounding box
-
-The `clip_box` function allows one to crop a raster by the
-min/max of the x and y coordinates. Note that we are cropping the original image `raster` now, and not the overview image `raster_overview`.
+Let's check the extent of the assets to find out its rough location in the raster image.
+We can use the `total_bounds` attribute to get the bounding box:
 
 ```python
-# Crop the raster with the bounding box
-raster_clip_box = raster.rio.clip_box(*fields.total_bounds)
-print(raster_clip_box.shape)
+assets.total_bounds
 ```
 
 ```output
-(3, 1565, 1565)
+array([27.7121001 , 35.87837949, 28.24591124, 36.45725024])
 ```
 
+The bounding box is in the form of `[minx, miny, maxx, maxy]`. Comparing this values with the raster image, we can see the magnitude of the bounding box coordinates does not match the coordinates of the raster image. This is because the two datasets have different coordinate reference systems (CRS). Therefore, we need to align the CRS of the two datasets before cropping the raster image.
 
-We successfully cropped the raster to a much smaller piece. We can visualize it now:
+Considering the raster image has larger data volume than the vector data, we will reproject the vector data to the CRS of the raster data. We can use the `to_crs` method:
 
 ```python
-raster_clip_box.plot.imshow(figsize=(8,8))
-```
-![](fig/E08/crop-raster-crop-by-bb-02.png){alt="Raster cropped by a bounding box"}
+# Reproject
+assets = assets.to_crs(visual.rio.crs)
 
-This cropped image can be saved for later usage:
-```python
-raster_clip_box.rio.to_raster("raster_clip.tif")
+# Check the new bounding box
+assets.total_bounds
 ```
 
+```output
+array([ 564058.0257114, 3970719.4080227,  611743.71498815, 4035358.56340039])
+```
 
-## Crop raster data with polygons
-
-We have a cropped image around the fields. To further analyze the fields, one may want to crop the image to the exact field boundaries.
-This can be done with the `clip` function:
+Now the bounding box coordinates are updated. We can use the `clip_box` function to crop the raster image to the bounding box of the vector data:
 
 ```python
-raster_clip_fields = raster_clip_box.rio.clip(fields['geometry'])
+# Crop the raster with the bounding box
+visual_clipbox = visual.rio.clip_box(*assets.total_bounds)
+
+# Visualize the cropped image
+visual_clipbox.plot.imshow()
 ```
 
+![](fig/E08/visual_clip_box.png){alt="Clip box results"}
 
-And we can visualize the results:
-```python
-raster_clip_fields.plot.imshow(figsize=(8,8))
-```
-![](fig/E08/crop-raster-crop-fields.png){alt="Ratser cropped by field polygons"}
+:::callout
+## Code Tip
+Cropping a raster with a bounding box is a quick way to reduce the size of the raster data. Since this operation is based on min/max coordinates, it is not as expensive as cropping with polygons, which requires more accurate overlay operations.
```

Comment:
```suggestion
Cropping a raster with a bounding box is a quick way to reduce the size of the raster data. Since this operation is based on min/max coordinates, it is not as computational extensive as cropping with polygons, which requires more accurate overlay operations.
```